### ISSUE TITLE
`set-clipboard off` option is not honored when running `load-buffer -w`

### ISSUE THREAD

**Issue by thinety** [tmux/tmux#3858]
### Issue description

`set-clipboard off` option doesn't have the expected behavior (should not affect system clipboard) when running `load-buffer -w`.

From the manpage:
> If set to on, tmux will both accept the escape sequence to create a buffer and attempt to set the terminal clipboard. If set to external, tmux will attempt to set the terminal clipboard but ignore attempts by applications to set tmux buffers. **If off, tmux will neither accept the clipboard escape sequence nor attempt to set the clipboard.**

### How to reproduce

```bash
# inside tmux

./tmux set-option -s set-clipboard on
echo "foo" | ./tmux load-buffer -w -
# system clipboard is now "foo"

./tmux set-option -s set-clipboard external
echo "bar" | ./tmux load-buffer -w -
# system clipboard is now "bar"

./tmux set-option -s set-clipboard off
echo "baz" | ./tmux load-buffer -w -
# system clipboard is now "baz", expected to still be "bar"
```

### Required information

* Tmux version: `tmux next-3.5` (latest master)
* Platform: Arch Linux
* `$TERM` outside: `alacritty`
* `$TERM` inside: `tmux-256color`

### Possible solution

The change below makes `tmux load-buffer -w` behave as expected. I can submit a PR if necessary.

```diff
diff --git a/cmd-load-buffer.c b/cmd-load-buffer.c
index 70fd7ed9..f40b72a9 100644
--- a/cmd-load-buffer.c
+++ b/cmd-load-buffer.c
@@ -100,7 +100,8 @@ cmd_load_buffer_exec(struct cmd *self, struct cmdq_item *item)
        cdata->item = item;
        if (bufname != NULL)
                cdata->name = xstrdup(bufname);
-       if (args_has(args, 'w') && tc != NULL) {
+       if (args_has(args, 'w') && tc != NULL &&
+               options_get_number(global_options, "set-clipboard") != 0) {
                cdata->client = tc;
                cdata->client->references++;
        }
```

**Comment by nicm** [comment]
The only reason to use `-w` is to set the clipboard, what's the point in requiring `set-clipboard` to be on? Just don't pass `-w`?

**Comment by thinety** [comment]
I've actually run into this in a more involved situation: by default, Neovim detects when it is inside tmux and uses `tmux load-buffer -w -` to integrate the `"+` and `"*` registers with the system clipboard.

The thing is that, sometimes, I do not want Neovim to be able to mess with the system clipboard, specially when connecting via `ssh`. So I added `set-option -s set-clipboard off` to tmux's config file in the remote system and, to my surprise, that didn't work. I've boiled it down to the behavior described in this issue.
